src/v2/Apps/Fair/Routes/FairExhibitors.tsx

```diff
@@ -55,61 +110,113 @@ const FairExhibitors: React.FC<FairExhibitorsProps> = ({ fair, relay }) => {
       }
 
       if (err) {
-        console.error(err)
+        logger.error(err)
       }
     })
   }
 
   return (
     <>
-      {/* @ts-expect-error STRICT_NULL_CHECK */}
-      {fair.exhibitors?.edges.map(({ node: show }, index) => {
-        if (show.counts.artworks === 0 || !show.partner) {
-          // Skip rendering of booths without artworks
-          return null
-        }
+      <Media at="xs">
+        <Sticky>
+          {({ stuck }) => {
+            return (
+              <Flex
+                justifyContent="flex-end"
+                py={1}
+                {...(stuck
+                  ? {
+                      px: 2,
+                      borderBottom: "1px solid",
+                      borderColor: "black10",
+                    }
+                  : {})}
+              >
+                <FairExhibitorSortFilter />
+              </Flex>
+            )
+          }}
+        </Sticky>
+      </Media>
+
+      <Media greaterThan="xs">
+        <Flex justifyContent="flex-end">
+          <FairExhibitorSortFilter />
+        </Flex>
+      </Media>
+
+      <Spacer mt={4} />
+
+      <LoadingArea isLoading={isGridLoading}>
+        {fair.exhibitors?.edges!.map((edge, index) => {
+          const show = edge?.node!
+          if (show?.counts?.artworks === 0 || !show?.partner) {
+            // Skip rendering of booths without artworks
+            return null
+          }
 
-        return (
-          <Box my={6} key={index}>
-            <FairExhibitorRail key={show.id} show={show} />
-          </Box>
-        )
-      })}
-
-      <Grid my={6}>
-        <Row>
-          <Col sm={6} mx="auto">
-            <Button
-              width="100%"
-              variant="secondaryGray"
-              onClick={handleClick}
-              loading={isLoading}
-              disabled={!relay.hasMore()}
-            >
-              Show more
-            </Button>
-          </Col>
-        </Row>
-      </Grid>
+          return (
+            <Box my={6} key={index}>
+              <FairExhibitorRail key={show.id} show={show} />
+            </Box>
+          )
+        })}
+      </LoadingArea>
+
+      <Spacer mt={4} />
+
+      <GridColumns>
+        <Column span={6} start={4}>
+          <Button
+            width="100%"
+            variant="secondaryGray"
+            onClick={handleClick}
+            loading={isButtonLoading}
+            disabled={!relay.hasMore()}
+          >
+            Show more
+          </Button>
+        </Column>
+      </GridColumns>
     </>
   )
 }
 
+const FairExhibitorsWithContext: React.FC<FairExhibitorsProps> = ({
+  ...props
+}) => {
+  const {
+    match: { location },
+  } = useRouter()
+  return (
+    <ExhibitorFilterContextProvider
+      filters={location.query}
+      sortOptions={[
+        { text: "Relevance", value: "FEATURED_DESC" },
+        { text: "Alphabetical (A-Z)", value: "NAME_ASC" },
```

Comment:
[this line](https://github.com/artsy/gravity/blob/f3748343a32d85ec2afc9817e287b3964fa7667d/app/api/v1/fairs_shows_endpoint.rb#L27) adds `"sortable"` to the string start in case the string contains `"name"`. So if we pass `SORTABLE_NAME_ASC` an error occurs (screenshot below). But if we pass `NAME_ASC` gravity turns it into `"sortable_name"`

![Screenshot 2021-07-19 at 10 27 55](https://user-images.githubusercontent.com/44819355/126120196-f8990c58-339f-4867-b3bc-2f6348da0c0f.png)

Reply:
Ahh, strange! Thanks for the context.